test/Microsoft.TestPlatform.AcceptanceTests/Extension/CompatibilityRowsBuilder.cs

```diff
@@ -0,0 +1,417 @@
+            TargetFramework = hostFramework,
+            InIsolationValue = inIsolation ? AcceptanceTestBase.InIsolation : null
+        };
+    }
+
+    private DllInfo GetMSTestInfo(string msTestVersion)
+    {
+        var depsXml = GetDependenciesXml();
+
+        // It is okay when node is null, we check that Version has value when we update paths by using MSTestInfo, and throw.
+        // This way it throws in the body of the test which has better error reporting than throwing in the data source.
+        XmlNode? node = depsXml.DocumentElement?.SelectSingleNode($"PropertyGroup/MSTestFramework{msTestVersion}Version");
+        var version = node?.InnerText.Replace("[", "").Replace("]", "");
+        var slash = Path.DirectorySeparatorChar;
+        var versionSpecificBinPath = $"{slash}bin{slash}MSTest{msTestVersion}-{version}{slash}";
+
+        return new DllInfo
+        {
+            Name = "MSTest",
+            PropertyName = "MSTest",
+            VersionType = msTestVersion,
+            Version = version,
+            Path = versionSpecificBinPath,
+        };
+    }
+
+    private static VSTestConsoleInfo GetVSTestConsoleInfo(string vstestConsoleVersion, RunnerInfo runnerInfo)
+    {
+        var depsXml = GetDependenciesXml();
+
+        // When version is Latest, we built it locally, but it gets restored into our nuget cache on build
+        // same as other versions, we just need to grab the version from a different property. 
+
+        var propertyName = vstestConsoleVersion == AcceptanceTestBase.LATEST
+            ? $"NETTestSdkVersion"
+            : $"VSTestConsole{vstestConsoleVersion}Version";
+
+        var packageName = runnerInfo.IsNetFrameworkRunner
+            ? "microsoft.testplatform"
+            : "microsoft.testplatform.cli";
+
+        // It is okay when node is null, we will fail to find the executable later, and throw.
+        // This way it throws in the body of the test which has better error reporting than throwing in the data source.
+        // And we can easily find out what is going on because --WRONG-VERSION-- sticks out, and is easy to find in the codebase.
+        XmlNode? node = depsXml.DocumentElement?.SelectSingleNode($"PropertyGroup/{propertyName}");
+        var version = node?.InnerText.Replace("[", "").Replace("]", "") ?? "--WRONG-VERSION--";
+        var vstestConsolePath = runnerInfo.IsNetFrameworkRunner
+            ? Path.Combine(IntegrationTestEnvironment.TestPlatformRootDirectory, "packages", packageName, version,
+                "tools", "net451", "Common7", "IDE", "Extensions", "TestPlatform", "vstest.console.exe")
+            : Path.Combine(IntegrationTestEnvironment.TestPlatformRootDirectory, "packages", packageName, version,
+                "contentFiles", "any", "netcoreapp2.1", "vstest.console.dll");
+
+        if (version.StartsWith("15."))
+        {
+            vstestConsolePath = vstestConsolePath.Replace("netcoreapp2.1", "netcoreapp2.0");
```

Comment:
Legacy version 15. is built for netcoreapp2.0 not netcoreapp2.1, so the location of the runner in the nuget package is different.

Terrible way of checking the version, but we control all of this code. And it will throw file not found, so it's easy to see where we went wrong.  Which is exactly how I added this if.